src/modules/ticket/actions.ts

```diff
@@ -0,0 +1,18 @@
+import {ThunkAction} from 'redux-thunk';
+import {RootAction, RootState} from '@store/rootReducer';
+import {buyTicketCall} from 'src/api/api';
+
+export const buy = (
+  flightId: string,
+): ThunkAction<void, RootState, unknown, RootAction> => (dispatch) => {
+  dispatch({type: 'BUY_TICKET_REQUEST', payload: flightId});
+
+  buyTicketCall(flightId)
+    .then((response) => {
+      dispatch({
+        type: 'BUY_TICKET_SUCCESS',
+        payload: {reservationId: response.data.reservation_id},
+      });
+    })
+    .catch((error) => console.warn(error));
+};
```

Comment:
we can replace for: 

```suggestion
export const buy = (
  flightId: string,
): ThunkAction<void, RootState, unknown, RootAction> => async (dispatch) => {
dispatch(buyTicketRequest(flightId);

try {
  const { data: { reservation_id } } = await buyTicketCall(flightId)
  
  if (reservation_id) {
      dispatch(buyTicketSuccess(reservation_id);
    }
    
    } catch(error) => {
        buyTicketFail(error);
    };
```


because we already have the actions creators, so it better to use them instead write it twice :)